spyder/plugins/editor/widgets/editor.py

```diff
@@ -499,10 +500,21 @@ def __init__(self, parent, actions):
             _("Find symbols in file..."),
             icon=ima.icon('symbol_find'),
             triggered=self.open_symbolfinder_dlg)
-        copy_to_cb_action = create_action(self, _("Copy path to clipboard"),
-                icon=ima.icon('editcopy'),
-                triggered=lambda:
-                QApplication.clipboard().setText(self.get_current_filename()))
+        copy_absolute_path_action = create_action(
+            self,
+            _("Copy abosolute path"),
+            icon=ima.icon('editcopy'),
+            triggered=lambda:
+            QApplication.clipboard().setText(
+                self.get_current_filename()))
+        copy_relative_path_action = create_action(
+            self,
+            _("Copy relative path"),
+            icon=ima.icon('editcopy'),
+            triggered=lambda:
+            QApplication.clipboard().setText(
+                (osp.relpath(self.get_current_filename(),
+                             getcwd_or_home()).replace(os.sep, "/"))))

```

Comment:
Let's move this to a method called `copy_relative_path`.

However, this code could fail on Windows if you try to get the relative path of a file placed on `D:\` and the user directory is placed on `C:\`. @dalthviz, could you take a look at that?

Also, if the user is working in a project, the path should be relative to the project.

Reply:
Yep, checking trying to get the relative path will end up in an error (if doing it with a mismatch in the drives):

```python
 Traceback (most recent call last):
  File "C:\Users\Daniel\Documents\Spyder\Spyder otros\juanita\spyder\spyder\plugins\editor\widgets\editor.py", line 517, in <lambda>
    getcwd_or_home()).replace(os.sep, "/"))))
  File "C:\Users\Daniel\Anaconda3\envs\develop\lib\ntpath.py", line 562, in relpath
    path_drive, start_drive))
ValueError: path is on mount 'E:', start on mount 'C:'
```

An option could be to detect that the paths aren't in the same drive and in that case return the absolute path of the file. To get the drive of a path [`os.path.splitdrive`](https://docs.python.org/3/library/os.path.html#os.path.splitdrive) can be used

Reply:
@dalthviz, could you check how we are doing this in Files? I think we could refactor the code there are use it here too.

Reply:
I think in the Files pane this case doesn't happen since the working directory is in sync with the current directory you are exploring. Just in case, the code that handles the action there is here: https://github.com/spyder-ide/spyder/blob/f17e1c59312a75e37401e9874e2fccc091c95c4a/spyder/plugins/explorer/widgets/explorer.py#L1072-L1106